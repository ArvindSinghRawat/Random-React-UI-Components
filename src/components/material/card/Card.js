import React from 'react';

// Styles
import styles from './Card.module.css';

export const Card = (props) => {
  const cssClasses = [styles.Card, styles.ShadowSmooth];
  if (props.type) {
    const type = props.type.toLowerCase();
    switch (props.type) {
      case 'dark':
        cssClasses.push(styles.dark);
        break;
      case 'xdark':
        cssClasses.push(styles.xdark);
        break;
      default:
        console.log(type + ' type not found');
        break;
    }
  }
  return (
    <div
      className={cssClasses.join(' ') + ' ' + props.className}
      style={{ ...props.style }}
    >
      {props.children}
    </div>
  );
};
