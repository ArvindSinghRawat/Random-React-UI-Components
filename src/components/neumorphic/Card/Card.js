import React from 'react';
import PropTypes from 'prop-types';

// Styles
import styles from './Card.module.css';

const possibleType = ['dark', 'xdark', 'inset', 'inset-dark', 'inset-xdark'];

export const Card = (props) => {
  const cssClasses = [styles.Card];
  if (props.type) {
    const type = props.type.toLowerCase();
    switch (type) {
      case 'light':
        cssClasses.push(styles.light);
        break;
      case 'dark':
        cssClasses.push(styles.dark);
        break;
      case 'xdark':
        cssClasses.push(styles.xdark);
        break;
      case 'inset':
        cssClasses.push(styles.Inset);
        break;
      case 'inset-dark':
        cssClasses.push(styles.dark);
        cssClasses.push(styles.Inset);
        break;
      case 'inset-light':
        cssClasses.push(styles.light);
        cssClasses.push(styles.Inset);
        break;
      case 'inset-xdark':
        cssClasses.push(styles.xdark);
        cssClasses.push(styles.Inset);
        break;
      default:
        console.log(type + ' type not found');
        break;
    }
  }
  const finalClass =
    cssClasses.join(' ') + (props.className ? ' ' + props.className : '');
  return (
    <div className={finalClass} style={{ ...props.style }}>
      {props.children}
    </div>
  );
};

Card.propTypes = {
  type: PropTypes.oneOf(possibleType),
  className: PropTypes.string,
  style: PropTypes.object,
  children: PropTypes.oneOfType([PropTypes.element, PropTypes.node])
};
