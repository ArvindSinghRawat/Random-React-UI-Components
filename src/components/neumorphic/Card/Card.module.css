.Card {
  color: #000;
  box-shadow: 4px 4px 24px 0 #fafafa, -4px -4px 12px 0 #a3b1e6;
  border-radius: 10px;
}

.Card.light {
  background-color: '#E0E5EC';
  box-shadow: 4px 4px 24px 0 #fafafa, -4px -4px 12px 0 #a3b1e6;
}

.Card.dark {
  color: #fff;
  background-color: #255763;
  box-shadow: 4px 4px 24px #2e6d7c, -4px -4px 8px #1c414a;
}

.Card.xdark {
  color: #fff;
  background-color: #0a0a0a;
  box-shadow: 4px 4px 44px #040404, -4px -4px 8px #060606;
}

.Card.Inset {
  background-color: '#E0E5EC';
  box-shadow: inset 5px 5px 12px 0 #a3b1e6, inset -5px -5px 15px 0 #fafafa !important;
}

.Card.light.Inset {
  box-shadow: inset 5px 5px 12px 0 #a3b1e6, inset -5px -5px 15px 0 #fff !important;
}

.Card.dark.Inset {
  color: #fff;
  background-color: #255763;
  box-shadow: inset 4px 4px 24px #1a3e46, inset -4px -4px 8px #307080 !important;
}

.Card.xdark.Inset {
  color: #fff;
  background-color: #0a0a0a;
  box-shadow: inset 4px 4px 14px #0a0a0a, inset -4px -4px 18px #010101 !important;
}
