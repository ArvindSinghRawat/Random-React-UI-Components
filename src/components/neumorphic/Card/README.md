# Neumorphic Card
## Introduction
This is a wrapper that allows to wrap content with a component that gives look and feel of contour of surface.

Each Card comes with inset and outset classes. 
> - __*Default class*__ resembles a surface protruding from the base.
> - __*Inset class*__ resembles a surface sunken in the base.
>   - Add ```inset-``` in front of desired type attribute.

#### Image of a Neumorphic Card

> ![Image of Neumorphic Card in Action](https://github.com/ArvindSinghRawat/Random-React-UI-Components/blob/master/src/components/neumorphic/Card/screenshots/DefaultNeumorphicCard.png?raw=true)

## Notes
- It is highly recommended to use the same background colour as colour shades on Card.
- User can also assign their own styles, to create unique Neumorphic Card.
- This effect is created by adding two differently coloured shadows on diagonally opposite corners.
- One of the corner should have light coloured shadow as compared to the Background while other one should have darker shadow.

## Types
Neumorphic Card comes with four types : 
- __*Default*__ (Colour: White);
- __*Light*__ (Colour: LightCyan(#e0e5ec)); 
- __*Dark*__ (Colour: DarkCyan(#255763)); 
- and __*Extra Dark*__ (Colour: Black).

### Default

#### Default type

> Usage
    
```javascript
    <NeumorphicCard>
        This is Neumorphic Card
    </NeumorphicCard>
```
> Result 
> ![Image of Neumorphic Card](https://github.com/ArvindSinghRawat/Random-React-UI-Components/blob/master/src/components/neumorphic/Card/screenshots/DefaultNeumorphicCard.png?raw=true)

#### Inset type

> Usage
    
```javascript
    <NeumorphicCard type="inset">
        This is Neumorphic Card
    </NeumorphicCard>
```
> Result 
> ![Image of Neumorphic Card](https://github.com/ArvindSinghRawat/Random-React-UI-Components/blob/master/src/components/neumorphic/Card/screenshots/InsetDefaultNeumorphicCard.png?raw=true)

### Light

#### Default
> Usage
    
```javascript
    <NeumorphicCard
        type='light'
    >
        This is Neumorphic Card
    </NeumorphicCard>
```
> Result 
> ![Image of Neumorphic Card](https://github.com/ArvindSinghRawat/Random-React-UI-Components/blob/master/src/components/neumorphic/Card/screenshots/LightNeumorphicCard.png?raw=true)

#### Inset

> Usage
    
```javascript
    <NeumorphicCard
        type='inset-light'
    >
        This is Neumorphic Card
    </NeumorphicCard>
```
> ![Image of Neumorphic Card](https://github.com/ArvindSinghRawat/Random-React-UI-Components/blob/master/src/components/neumorphic/Card/screenshots/InsetLightNeumorphicCard.png?raw=true)

### Dark

#### Default
> Usage
    
```javascript
    <NeumorphicCard
        type='dark'
    >
        This is Neumorphic Dark Card
    </NeumorphicCard>
```
> Result 
> ![Image of Neumorphic Card](https://github.com/ArvindSinghRawat/Random-React-UI-Components/blob/master/src/components/neumorphic/Card/screenshots/DarkNeumorphicCard.png?raw=true)

#### Inset

> Usage
    
```javascript
    <NeumorphicCard
        type='inset-dark'
    >
        This is Neumorphic Dark Card
    </NeumorphicCard>
```
> ![Image of Neumorphic Card](https://github.com/ArvindSinghRawat/Random-React-UI-Components/blob/master/src/components/neumorphic/Card/screenshots/InsetDarkNeumorphicCard.png?raw=true)


### Extra Dark

#### Default
> Usage
    
```javascript
    <NeumorphicCard
        type='xdark'
    >
        This is Neumorphic Extra Dark Card
    </NeumorphicCard>
```
> Result 
> ![Image of Neumorphic Card](https://github.com/ArvindSinghRawat/Random-React-UI-Components/blob/master/src/components/neumorphic/Card/screenshots/ExtraDarkNeumorphicCard.png?raw=true)

#### Inset

> Usage

```javascript
    <NeumorphicCard
        type='inset-xdark'
    >
        This is Neumorphic Extra Dark Card
    </NeumorphicCard>
```
> ![Image of Neumorphic Card](https://github.com/ArvindSinghRawat/Random-React-UI-Components/blob/master/src/components/neumorphic/Card/screenshots/InsetExtraDarkNeumorphicCard.png?raw=true)



- __* Extra styling of positioning and margin is added to increase size of card.__

- __* By Default, Card will closely wrap the child component.__

- Any valid react component can be wrapped in this Card.
