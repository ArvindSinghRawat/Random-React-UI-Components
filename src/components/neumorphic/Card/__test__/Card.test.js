import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { Card, possibleType } from '../Card';

configure({ adapter: new Adapter() });

describe('<Card /> : ', () => {
  let wrapper = null;
  const content = 'This is a Card.';
  beforeEach(() => {
    wrapper = shallow(<Card>{content}</Card>);
  });

  // Testing for type props

  it('Should correctly render <Card /> for each type', () => {
    expect(wrapper.text()).toMatch(content);
    expect(wrapper).toMatchSnapshot();
    for (const type of possibleType) {
      wrapper.setProps({ type: type });
      expect(wrapper.text()).toMatch(content);
      expect(wrapper).toMatchSnapshot();
    }
  });

  it('Should correctly render <Card/> with console.log() error of Unknown type', () => {
    console.log = jest.fn();
    console.error = jest.fn();
    const type = 'unknown';
    wrapper.setProps({ type: type });
    expect(console.log).toHaveBeenCalledWith(type + ' type not found');
    expect(console.error).toHaveBeenCalled();
  });

  it('Should correctly render <Card className="customClass" />', () => {
    wrapper.setProps({ className: 'customClass' });
    expect(wrapper.text()).toMatch(content);
    expect(wrapper).toMatchSnapshot();
  });
});
