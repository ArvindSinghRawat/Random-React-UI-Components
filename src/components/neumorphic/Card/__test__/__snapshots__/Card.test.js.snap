// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> :  Should correctly render <Card /> for each type 1`] = `
<div
  className="Card"
  style={Object {}}
>
  This is a Card.
</div>
`;

exports[`<Card /> :  Should correctly render <Card className="customClass" /> 1`] = `
<div
  className="Card customClass"
  style={Object {}}
>
  This is a Card.
</div>
`;

exports[`<Card /> :  Should correctly render <Card type="dark" /> for each type 1`] = `
<div
  className="Card dark"
  style={Object {}}
>
  This is a Card.
</div>
`;

exports[`<Card /> :  Should correctly render <Card type="inset" /> for each type 1`] = `
<div
  className="Card Inset"
  style={Object {}}
>
  This is a Card.
</div>
`;

exports[`<Card /> :  Should correctly render <Card type="inset-dark" /> for each type 1`] = `
<div
  className="Card dark Inset"
  style={Object {}}
>
  This is a Card.
</div>
`;

exports[`<Card /> :  Should correctly render <Card type="inset-light" /> for each type 1`] = `
<div
  className="Card light Inset"
  style={Object {}}
>
  This is a Card.
</div>
`;

exports[`<Card /> :  Should correctly render <Card type="inset-xdark" /> for each type 1`] = `
<div
  className="Card xdark Inset"
  style={Object {}}
>
  This is a Card.
</div>
`;

exports[`<Card /> :  Should correctly render <Card type="light" /> for each type 1`] = `
<div
  className="Card light"
  style={Object {}}
>
  This is a Card.
</div>
`;

exports[`<Card /> :  Should correctly render <Card type="xdark" /> for each type 1`] = `
<div
  className="Card xdark"
  style={Object {}}
>
  This is a Card.
</div>
`;
